clusterman/kubernetes/kubernetes_cluster_connector.py

```diff
@@ -129,29 +129,46 @@ def get_unschedulable_pods(
             unschedulable_pods.append((pod, self._get_pod_unschedulable_reason(pod)))
         return unschedulable_pods
 
-    def freeze_agent(self, agent_id: str) -> None:
+    def freeze_agent(self, node_name: str) -> None:
         now = str(arrow.now().timestamp)
         try:
             body = {
                 "spec": {"taints": [{"effect": "NoSchedule", "key": CLUSTERMAN_TERMINATION_TAINT_KEY, "value": now}]}
             }
-            self._core_api.patch_node(agent_id, body)
+            self._core_api.patch_node(node_name, body)
         except ApiException as e:
-            logger.warning(f"Failed to freeze {agent_id}: {e}")
+            logger.warning(f"Failed to freeze {node_name}: {e}")
 
-    def drain_node(self, agent_id: str) -> bool:
+    def clean_node(self, node_name: str) -> bool:
         try:
-            logger.info(f"Cordoning {agent_id}...")
-            if not self.cordon_node(agent_id):
+            logger.info(f"Cordoning {node_name}...")
+            self.cordon_node(node_name)
+        except Exception as e:
+            logger.warning(f"Failed to cordon {node_name} continuing to delete pods anyway: {e}")
+        try:
+            logger.info(f"Deleting pods on {node_name}...")
+            if not self._delete_tasks_from_node(node_name):
+                logger.info("Some pods couldn't be deleted")
                 return False
-            logger.info(f"Evicting pods on {agent_id}...")
-            if not self._evict_tasks_from_node(agent_id):
+            logger.info(f"Cleaned {node_name}")
+            return True
+        except Exception as e:
+            logger.warning(f"Failed to clean {node_name}: {e}")
+            return False
+
+    def drain_node(self, node_name: str) -> bool:
+        try:
+            logger.info(f"Cordoning {node_name}...")
+            if not self.cordon_node(node_name):
+                return False
+            logger.info(f"Evicting pods on {node_name}...")
+            if not self._evict_tasks_from_node(node_name):
                 logger.info("Some pods couldn't be evicted")
```

Comment:
on mobile right now so I can't just do a GH suggestion - but thoughts on having this include the node name as well?